acid/features/zuul_connector/connector.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+from contextlib import contextmanager
+import shlex
+import os
+
+import paramiko
+
+from .exceptions import ZuulConnectorFailed
+
+
+class ZuulConnector:
+    def __init__(self, host, username, user_key_file, host_keys_file,
+                 tenant, trigger, project):
+        self.user_key = paramiko.RSAKey.from_private_key_file(user_key_file)
+        self.host = host
+        self.username = username
+        self.host_pub_key = host_keys_file
+
+        self.tenant = tenant
+        self.trigger = trigger
+        self.project = project
+        self._client = self._prepare_client(self.host_pub_key)
+
+    @contextmanager
+    def ssh_connection(self):
+        try:
+            self._client.connect(hostname=self.host, username=self.username,
+                                 pkey=self.user_key)
+            yield self._client
+        except paramiko.ssh_exception.SSHException as exc:
+            raise ZuulConnectorFailed(f"{type(exc).__name__}: {exc}")
+        finally:
+            self._client.close()
+
+    def enqueue(self, pipeline, branch):
+        command = f"zuul enqueue-ref --tenant { shlex.quote(self.tenant) }" \
+            f" --trigger { shlex.quote(self.trigger) }" \
+            f" --pipeline { shlex.quote(pipeline) }" \
+            f" --project { shlex.quote(self.project) }" \
+            f" --ref { shlex.quote(f'refs/head/{branch}') } &"
+
+        with self.ssh_connection() as connection:
+            connection.exec_command(command)  # noqa S601
+
+    def dequeue(self, pipeline, branch):
+        command = f"zuul dequeue --tenant { shlex.quote(self.tenant) }" \
```

Comment:
Try to avoid trailing slashes to break string on multiple lines. It is easy to omit slash and whole string becomes tuple. This bug is very hard to spot. Use `str` constructor instead.
```python
command = str(f"zuul enqueue-ref --tenant {shlex.quote(self.tenant)}" 
              f" --trigger {shlex.quote(self.trigger)}"
              f" --pipeline {shlex.quote(pipeline)}"
              f" --project {shlex.quote(self.project)}"
              f" --ref {shlex.quote(f'refs/head/{branch}')} &")
```
And also don't use spaces with brackets inside f-strings.
Instead of `{ shlex.quote(self.trigger) }` do `{shlex.quote(self.trigger)}`